test/e2e/test_cluster_configs/storage_profile.json

```diff
@@ -1,6 +1,7 @@
 {
 	"env": {
-		"CREATE_VNET": true
+		"CREATE_VNET": true,
+		"GINKGO_SKIP": "should create a pv by deploying a pod that consumes a pvc"
```

Comment:
You could also just edit that test case to only run if all nodes have managed disks.

`IsManagedDisks()` method of `AgentPoolProfile` should do the trick